### PR TITLE
Waiting for slave Activity, closes #121

### LINKED ISSUE #121: Waiting for slave to verify rhythm activity
- Parent issues: #109 
- Relevant issues: #116
- Estimated effort: 1 (Joost)

* * *

## User story
As a user I want the app to let me know that the client is mocking my rhythm so that I know the pairing is going on successfully

## What
This issue demands to add an (almost) empty activity that notifies the user that the slave is mocking the rhythm.

## Why
This issue is needed for the project because the user shouldn't exit the app because he thinks that the pairing has failed (because he's not paired securely yet)

## How
This feature can be added by adding an empty activity with as only views some informing text, a button to cancel the pairing and a button to make a new rhythm.

## Definition of done
This task can be considered done when

- There is an activity that shows the user that the master is busy creating the rhythm.
- There is a button to cancel the pairing and return to MainActivity
- There is a button to return to the rhythm creating activity

### PR DESCRIPTION
**Pull request by jverbraeken**
- Relevant Issues: #121 
- Related Pull Requests: -

* * *

## What
This Pull Request adds to the repository a screen that shows a nice waiting animation to let the user know that he should wait for his partner to do his work to finish the pairing.

## Why
This Pull Request is needed because the user should know that he should wait for the other user.

## How
This feature can be viewed/tested within the project by changing in EntryActivity LoginActivity to WaitForSlaveActivity

## Alternative implementation
Other implementations that I've have considered are -

### CODE REVIEW

File: app/src/main/java/com/nervousfish/nervousfish/activities/WaitForSlaveActivity.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+import com.nervousfish.nervousfish.service_locator.IServiceLocator;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Used to let the Bluetooth-initiating user know that he should wait for his partner
+ * to complete the pairing session.
+ */
+@SuppressWarnings("PMD")
```
Comment (ericcornelissen) [inline app/src/main/java/com/nervousfish/nervousfish/activities/WaitForSlaveActivity.java]: You shouldn't suppress all PMD errors on this class

File: app/src/main/java/com/nervousfish/nervousfish/activities/WaitForSlaveActivity.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+public final class WaitForSlaveActivity extends Activity {
+    private static final Logger LOGGER = LoggerFactory.getLogger("LoginActivity");
+    private IServiceLocator serviceLocator;
+
+    @Override
```
Comment (ericcornelissen) [inline app/src/main/java/com/nervousfish/nervousfish/activities/WaitForSlaveActivity.java]: Please add JavaDoc

File: app/src/main/java/com/nervousfish/nervousfish/activities/WaitForSlaveActivity.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    protected void onCreate(final Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        final Intent intent = getIntent();
+        this.serviceLocator = (IServiceLocator) intent.getSerializableExtra(ConstantKeywords.SERVICE_LOCATOR);
+        setContentView(R.layout.wait_for_slave);
+        LOGGER.info("WaitForSlaveActivity created");
+
+        final Button cancelButton = (Button) findViewById(R.id.cancelWaitForSlave);
+        final Context context = this;
```
Comment (ericcornelissen) [inline app/src/main/java/com/nervousfish/nervousfish/activities/WaitForSlaveActivity.java]: Why not add a separate method in the class to do this?

**Comment by jverbraeken** [comment]
@ericcornelissen Implemented all your suggestions!

File: app/src/main/java/com/nervousfish/nervousfish/activities/WaitForSlaveActivity.java [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+        super.onCreate(savedInstanceState);
+        final Intent intent = getIntent();
+        this.serviceLocator = (IServiceLocator) intent.getSerializableExtra(ConstantKeywords.SERVICE_LOCATOR);
+        setContentView(R.layout.wait_for_slave);
+        LOGGER.info("WaitForSlaveActivity created");
+
+        initCancelButton();
+    }
+
+    private void initCancelButton() {
```
Comment (ericcornelissen) [inline app/src/main/java/com/nervousfish/nervousfish/activities/WaitForSlaveActivity.java]: This is not what I meant 😅 I was asking why the `onClick` method itself wasn't its own method so you don't have to set a `final Context context = this;`

File: app/src/main/java/com/nervousfish/nervousfish/activities/WaitForSlaveActivity.java [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+        final Button cancelButton = (Button) findViewById(R.id.cancelWaitForSlave);
+        final Context context = this;
+        cancelButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(final View view) {
+                final Intent intent = new Intent(context, MainActivity.class);
+                intent.putExtra(ConstantKeywords.SERVICE_LOCATOR, serviceLocator);
+                context.startActivity(intent);
```
Comment (clenrock) [inline app/src/main/java/com/nervousfish/nervousfish/activities/WaitForSlaveActivity.java]: Make a small test for this? 🙉

File: app/src/main/java/com/nervousfish/nervousfish/activities/WaitForSlaveActivity.java (hunk 4 above)
Comment (jverbraeken) [inline app/src/main/java/com/nervousfish/nervousfish/activities/WaitForSlaveActivity.java]: Done!

File: app/src/main/java/com/nervousfish/nervousfish/activities/WaitForSlaveActivity.java (hunk 5 above)
Comment (jverbraeken) [inline app/src/main/java/com/nervousfish/nervousfish/activities/WaitForSlaveActivity.java]: Yeah, it's mty first Cucumber tests :P

File: app/src/main/java/com/nervousfish/nervousfish/activities/WaitForSlaveActivity.java [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+    @Override
+    protected void onCreate(final Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        final Intent intent = getIntent();
+        this.serviceLocator = (IServiceLocator) intent.getSerializableExtra(ConstantKeywords.SERVICE_LOCATOR);
+        setContentView(R.layout.wait_for_slave);
+        LOGGER.info("WaitForSlaveActivity created");
+    }
+
+    public void cancelWaitingForSlave(final View view) {
```
Comment (ericcornelissen) [inline app/src/main/java/com/nervousfish/nervousfish/activities/WaitForSlaveActivity.java]: Missing Javadoc 😉

File: app/src/main/java/com/nervousfish/nervousfish/activities/LoginActivity.java [hunk 7]
```diff
@@ -49,6 +48,7 @@ public void onClick(final View v) {
         });
     }
 
+    @SuppressWarnings("PMD.AvoidFinalLocalVariable")
```
Comment (ericcornelissen) [inline app/src/main/java/com/nervousfish/nervousfish/activities/LoginActivity.java]: Are you sure we want to suppress this warning here ?

File: app/src/androidTest/java/com/nervousfish/nervousfish/test/LoginActivitySteps.java [hunk 8]
```diff
@@ -56,7 +56,6 @@ public void iHaveALoginActivity() {
         intent.putExtra(ConstantKeywords.SERVICE_LOCATOR,
                 getCurrentActivity().getIntent().getSerializableExtra(ConstantKeywords.SERVICE_LOCATOR));
         getActivity().startActivity(intent);
-        assertTrue(getCurrentActivity() instanceof LoginActivity);
```
Comment (ericcornelissen) [inline app/src/androidTest/java/com/nervousfish/nervousfish/test/LoginActivitySteps.java]: Are you sure you don't want this check? I would like to have it...

Comment (jverbraeken) [inline app/src/androidTest/java/com/nervousfish/nervousfish/test/LoginActivitySteps.java]: You don't want this check because the activity isn't set to LoginActivity immediately, that's done asynchronously and often isn't finished yet at that line

Comment (ericcornelissen) [inline app/src/androidTest/java/com/nervousfish/nervousfish/test/LoginActivitySteps.java]: I disagree. If you accept that as a given, you can never be sure that the next step (which depends on that activity being active) will succeed, since it can be that the activity isn't loaded yet

File: app/src/main/java/com/nervousfish/nervousfish/activities/LoginActivity.java (hunk 7 above)
Comment (jverbraeken) [inline app/src/main/java/com/nervousfish/nervousfish/activities/LoginActivity.java]: Yes, because the dummypassword is there and you remove it in another pr. So it's not needed to remove dummypassword in this pr

File: app/src/main/java/com/nervousfish/nervousfish/service_locator/EntryActivity.java [hunk 9]
```diff
@@ -31,9 +32,8 @@ protected void onCreate(final Bundle savedInstanceState) {
 
         LOGGER.info("EntryActivity created");
 
-        final Intent intent = new Intent(this, LoginActivity.class);
+        final Intent intent = new Intent(this, WaitForSlaveActivity.class);
```
Comment (ericcornelissen) [inline app/src/main/java/com/nervousfish/nervousfish/service_locator/EntryActivity.java]: Could you revert this change 😄

File: app/src/main/res/values/colors.xml [hunk 10]
```diff
@@ -3,4 +3,6 @@
     <color name="colorPrimary">#0099cc</color>
     <color name="colorPrimaryDark">#0673aa</color>
     <color name="colorAccent">#808183</color>
+
+    <color name="red_button">#9A3334</color>
```
Comment (ericcornelissen) [inline app/src/main/res/values/colors.xml]: Could you change this name to something like `alert_red`?

File: app/src/main/res/values/strings.xml [hunk 11]
```diff
@@ -16,7 +16,11 @@
     <string name="delete_popup_no_recovery">You won\'t be able to recover this contact!</string>
     <string name="cancel">Cancel</string>
 
-    <string name="yes_delete">Yes, delete it!</string>
+    <string name="yes_delete">Yes,delete it!</string>
```
Comment (ericcornelissen) [inline app/src/main/res/values/strings.xml]: Could you revert this change
